collector/src/lib.rs

```diff
@@ -570,7 +570,11 @@ impl<V: vdaf::Collector> Collector<V> {
             .headers()
             .get(CONTENT_TYPE)
             .ok_or(Error::BadContentType(None))?;
-        if content_type != CollectionJobResp::<TimeInterval>::MEDIA_TYPE {
+        if let Ok(mime) = content_type.to_str()?.parse::<mime::Mime>() {
```

Comment:
As argued in `http_handlers.rs`, prefer `Result::map_err` over `if let Ok(...) = foo.bar() {...} else {...}`

Reply:
Resolved in https://github.com/divviup/janus/pull/3649/commits/2ad6be874daf12c1c284c1606d5bf74cb2ff8a4c